repository/src/main/java/org/apache/atlas/repository/store/graph/v2/DataProductInputsOutputsMigrationService.java

```diff
@@ -0,0 +1,97 @@
+package org.apache.atlas.repository.store.graph.v2;
+
+import org.apache.atlas.AtlasErrorCode;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.model.instance.AtlasEntity;
+import org.apache.atlas.model.instance.AtlasRelatedObjectId;
+import org.apache.atlas.repository.graph.GraphHelper;
+import org.apache.atlas.repository.graphdb.AtlasEdge;
+import org.apache.atlas.repository.graphdb.AtlasEdgeDirection;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.apache.atlas.repository.Constants.*;
+import static org.apache.atlas.repository.store.graph.v2.preprocessor.PreProcessorUtils.*;
+
+public class DataProductInputsOutputsMigrationService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DataProductInputsOutputsMigrationService.class);
+
+    private final EntityGraphRetriever entityRetriever;
+
+
+    private String productGuid;
+    private final TransactionInterceptHelper   transactionInterceptHelper;
+
+    public DataProductInputsOutputsMigrationService(EntityGraphRetriever entityRetriever, String productGuid, TransactionInterceptHelper transactionInterceptHelper) {
+        this.entityRetriever = entityRetriever;
+        this.transactionInterceptHelper = transactionInterceptHelper;
+        this.productGuid = productGuid;
+    }
+
+    public void migrateProduct() throws Exception {
+        try {
+            AtlasVertex productVertex = entityRetriever.getEntityVertex(this.productGuid);
+            if(productVertex == null) {
+                throw new AtlasBaseException(AtlasErrorCode.INSTANCE_GUID_NOT_FOUND, this.productGuid);
+            }
+
+            migrateAttr(productVertex);
+            commitChanges();
```

Comment:
call commit only if required
migrateAttr may return a flag whether commit is needed, based on this flag decide whether need to call commit 
Also, in case commit is skipped as migration is not needed, print that as info log